File: MatrixMultiplier.java
import java.util.*;

public class MatrixMultiplier {
	
	static int[][] matrixA;
	static int[][] matrixB;
	static int[][] matrixC;
	
	public MatrixMultiplier (int N) {
		
			matrixA = new int[N][N];
			matrixB = new int[N][N];
			matrixC = new int[N][N];
					
	}
	
	public void fillMatrixRandom() {
		

		Random rn = new Random();
		
		
		int i = 0;
		int j = 0;
		
		for (i=0; i<matrixA.length; i++) {
			
			for (j=0; j<matrixA.length; j++) {
				
				matrixA[i][j] = rn.nextInt(matrixA.length*10);
				matrixB[i][j] = rn.nextInt(matrixA.length*10);
				
			}
			
		}
	}
	
	
	public void printInputMatrices() {
		
		int i = 0;
		int j = 0;
		
		for (i=0; i<matrixA.length; i++) {
			
			for (j=0; j<matrixA.length; j++) {
				
				System.out.print(" " + matrixA[i][j] + " ");
				
			}
			
			System.out.println();
			
		}
		
		for (i=0; i<matrixA.length; i++) {
			
			for (j=0; j<matrixA.length; j++) {
				
				System.out.print(" " + matrixB[i][j] + " ");
				
			}
			
			System.out.println();
			
		}
	}
	
	public void multiplyMatrices() {
		
		int i = 0;
		int j = 0;
		int k = 0;
		
		for (i=0; i<matrixA.length; i++) {
			
			for (j=0; j<matrixA.length; j++) {
				
				for (k=0; k<matrixA.length; k++) {
					
					matrixC[i][j] += matrixA[i][k] * matrixB[k][j];
					
				}
				
				
			}
			
			
		}
		
		
		
	}
	
	public void printResultMatrix() {
		
		int i = 0;
		int j = 0;
		
		for (i=0; i<matrixA.length; i++) {
			
			for (j=0; j<matrixA.length; j++) {
				
				System.out.print(" " + matrixC[i][j] + " ");
				
			}
			
			System.out.println();
			
		}
		
		
		
	}
	
	

	public static void main(String[] args) {
		

		MatrixMultiplier multiplier = new MatrixMultiplier(Integer.parseInt(args[0]));
		multiplier.fillMatrixRandom();
		//multiplier.printInputMatrices();
		multiplier.multiplyMatrices();
		//multiplier.printResultMatrix();
	
		

	}

}

File: sankalp/MatrixMultiplier.py
import sys
import random
class MatrixMultiplier:
	matrixA = [[]]
	matrixB = [[]]
	matrixC = [[]]

	matrixDim = 0
	randomMaxValue = 100
	def __init__(self, N):
		self.matrixDim = int(N)

		self.matrixA = [[0 for i in range(self.matrixDim)] for j in range(self.matrixDim)]
		self.matrixB = [[0 for i in range(self.matrixDim)] for j in range(self.matrixDim)]
		self.matrixC = [[0 for i in range(self.matrixDim)] for j in range(self.matrixDim)]

	def fillMatrices(self):
		
		i = 0
		j = 0

		for i in range (0, self.matrixDim):
			for j in range (0, self.matrixDim):

				self.matrixA[i][j] = random.randint(0, self.randomMaxValue)
				self.matrixB[i][j] = random.randint(0, self.randomMaxValue)

	
	def printInputMatrices(self):
		i = 0
		j = 0

		for i in range (0, self.matrixDim):
			for j in range (0, self.matrixDim):
				print " " + str(self.matrixA[i][j]) + " ",
			
			print "\n"

		for i in range (0, self.matrixDim):
			for j in range (0, self.matrixDim):

				print " " + str(self.matrixB[i][j]) + " ",
			
			print "\n"	

	def multiplyMatrices(self):
		
		for i in range(0, self.matrixDim):
			for j in range(0, self.matrixDim):
				for k in range (0, self.matrixDim):
					self.matrixC[i][j] += self.matrixA[i][k] * self.matrixB[k][j] 		


	def printOutputMatrix(self):

		for i in range (0, self.matrixDim):
			for j in range (0, self.matrixDim):

				print " " + str(self.matrixC[i][j]) + " ",
			
			print "\n"	



def main():
	multiplier = MatrixMultiplier(str(sys.argv[1]))
	multiplier.fillMatrices()
	#multiplier.printInputMatrices()
	multiplier.multiplyMatrices()
	#multiplier.printOutputMatrix()

if __name__ == "__main__":
	main()
